docs/intent_conversation_api.md

```diff
@@ -0,0 +1,230 @@
+---
+title: "Conversation API"
+sidebar_label: "Conversation API"
+---
+
+Intents can be recognized from text and fired using the [conversation integration](https://www.home-assistant.io/integrations/conversation/).
+
+An HTTP API endpoint is available at `/api/conversation/process`, which receives an [input sentence](#input-sentence) and produces an [intent response](#intent-response).
+A "conversation" is tracked across multiple inputs and responses using a [conversation id](#conversation-id) generated by Home Assistant.
+
+## Input Sentence
+
+A sentence may be POST-ed to `/api/conversation/process` like:
+
+```json
+{
+  "text": "turn on the lights in the living room",
+  "language": "en"
+}
+```
+
+The following input fields are available:
+
+| Name              | Type   | Description                                                                                 |
+|-------------------|--------|---------------------------------------------------------------------------------------------|
+| `text`            | string | Input sentence.                                                                             |
+| `language`        | string | Optional. Language of the input sentence (defaults to configured language).                 |
+| `conversation_id` | string | Optional. Unique id to [track conversation](#conversation-id). Generated by Home Assistant. |
+
+
+## Conversation Response
+
+The JSON response from `/api/conversation/process` contains information about the effect of the fired intent, for example:
+
+```json
+{
+  "response": {
+    "response_type": "action_done",
+    "language": "en",
+    "data": {
+      "targets": [
+        {
+          "type": "area",
+          "name": "Living Room"
+          "id": "living_room"
+        },
+        {
+          "type": "domain",
+          "name": "light"
+          "id": "light"
+        }
+      ]
+    },
+    "speech": {
+      "plain": {
+        "speech": "Turned Living Room lights on"
+      }
+    }
+  },
+  "conversation_id": "<generated-id-from-ha>",
+}
+```
+
+The following properties are available in the `"response"` object:
+
+| Name              | Type       | Description                                                                                         |
+|-------------------|------------|-----------------------------------------------------------------------------------------------------|
+| `response_type`   | string     | One of `action_done`, `query_answer`, or `error` (see [response types](#response-types)).           |
+| `data`            | dictionary | Relevant data for each [response type](#response_types).                                            |
+| `language`        | string     | Optional. The language of the intent and response.                                                  |
```

Comment:
This is not optional.

Reply:
I need to enforce this in the IntentResponse.